code/modules/buildmode/submodes/forcemove.dm

```diff
@@ -0,0 +1,55 @@
+// This tool calls forceMove() on an object/mob that is selected with right-click.
+// Once the atom is moved, its selection is cleared to avoid any further moves by accident.
+/datum/buildmode_mode/forcemove
+	key = "forcemove"
+	var/atom/selected_atom		// The atom we want to move
```

Comment:
this needs to have a signal registered on it to ensure that references are properly cleared if the atom is `qdel`'d

Reply:
Or just a soft ref